go/sql/model.go

```diff
@@ -0,0 +1,34 @@
+package sql
+
+// Key shared by tables
+type Key struct {
+	UniqueID string `json:"uniqueId" gorm:"primaryKey;column:uniqueId"`
+	UserID   uint   `json:"userId" gorm:"primaryKey;column:userId"`
+}
+
+// Model defines the interface for table and row
+type Model interface {
+	TableName() string
+	NewRow() Model
+	NewRows() interface{}
+	SetKey(Key)
+	GetKey() Key
+}
```

Comment:
How does this interface work - is Model a table or a row?

Reply:
It should be row.  I've updated it to be more consistent.